MaloWBotLK/Functions.lua

```diff
@@ -292,6 +296,59 @@ function mb_getMySpecName()
 	return name
 end
 
+-- Scans through all bag slots looking for item by name
+function mb_GetItemLocation(itemName)
+	for bag = 0, 4 do
+		for slot = 1, GetContainerNumSlots(bag) do
+			local texture = GetContainerItemInfo(bag, slot)
+			if texture ~= nil then
+				local name = GetItemInfo(mb_GetItemStringFromItemLink(GetContainerItemLink(bag, slot)))
+				if itemName == name then
+					return bag, slot
+				end
+			end
+		end
+	end
+	return nil
+end
 
+-- Converts an ItemLink to an ItemString
+function mb_GetItemStringFromItemLink(itemLink)
+	local found, _, itemString = string.find(itemLink, "^|%x+|H(.+)|h%[.+%]")
+	return itemString
+end
 
+-- Uses item in bag by name
+function mb_UseItem(itemName)
+	local bag, slot = mb_GetItemLocation(itemName)
+	if bag ~= nil then
+		UseContainerItem(bag, slot)
+		return true
+	end
+	return false
+end
+-- Uses mage food if missing mana and not in combat
+function mb_DrinkIfGood()
```

Comment:
Would probably rename this to just mb_Drink(), in general trying to name functions that checks if something should be done and then also doing it is really hard. Normally you'd want to split it into one "check" function and one "do" function. But, this project has a lot of these functions that have both checks and do's in them, using return true/false depending on whether something was done, so I think for this project it's expected that function like these only do things if it's good, so there for the "IfGood" part is a bit superfluous.